# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1282645137 -7200
#      Tue Aug 24 12:18:57 2010 +0200
# Node ID c98a0d05b24a27a0915381c71505d566ccf73688
# Parent  6105ed2db9d740cd05d993cb0d52baf81c94085d
inline char sorts for small boost to sortrows of char matrices

diff --git a/liboctave/Array-ch.cc b/liboctave/Array-ch.cc
--- a/liboctave/Array-ch.cc
+++ b/liboctave/Array-ch.cc
@@ -24,16 +24,18 @@ along with Octave; see the file COPYING.
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 // Instantiate Arrays of char values.
 
 #include "Array.h"
 #include "Array.cc"
+#define INLINE_ASCENDING_SORT
+#define INLINE_DESCENDING_SORT
 #include "oct-sort.cc"
 
 INSTANTIATE_ARRAY_SORT (char);
 
 INSTANTIATE_ARRAY (char, OCTAVE_API);
 
 template OCTAVE_API std::ostream& operator << (std::ostream&, const Array<char>&);
 
diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,12 @@
+2010-08-24  Jaroslav Hajek  <highegg@gmail.com>
+
+	* Array-ch.cc: Inline basic sorts.
+
 2010-08-21  Rik <octave@nomad.inbox5.com>
 
 	* CmplxQRP.cc, CmplxQRP.h, dbleQRP.cc, dbleQRP.h, fCmplxQRP.cc,
 	fCmplxQRP.h, floatQRP.cc, floatQRP.h: Declare Pvec as RowVector,
 	not ColumnVector.  Bug #30832.
 
 2010-08-18  Jaroslav Hajek  <highegg@gmail.com>
 
